### PR TITLE
Add From Source Test Run to Live Tests

### LINKED ISSUE #14918: Add a From Source Live Testing Pipeline
Add a `From Source` pipeline to live testing as we have for PR validation. This will help catch additional issues that only crop up when certain code paths emitted during playback testing are used.

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #14918 

This PR adds a From Source test run to live test runs. Adding the live test run adds parity to the live testing as we have with PR testing.

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -103,16 +109,36 @@ jobs:
...
+        displayName: 'Build and Install, JDK Version: $(JavaTestVersion)'
         inputs:
           mavenPomFile: pom.xml
           goals: 'install'
-          options: '$(DefaultOptions) -DskipTests -Dgpg.skip -Dmaven.javadoc.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true -Drevapi.skip=true -pl $(ProjectList) -am'
+          options: '$(DefaultOptions) -T 1C -DskipTests -Dgpg.skip -Dmaven.javadoc.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true -Drevapi.skip=true -pl $(ProjectList) -am'
           mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
           javaHomeOption: 'JDKVersion'
-          jdkVersionOption: $(JavaBuildVersion)
+          jdkVersionOption: $(JavaTestVersion)
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:141]: Was this broken before?

**Review by weshaggard — APPROVED** [review]
The change looks reasonable. I think you should do some extra test runs to ensure this new matrix entry works in the different service live test pipelines.

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml (hunk 1 above)
Comment (alzimmermsft) [reply eng/pipelines/templates/jobs/archetype-sdk-tests.yml:141]: We were always building with Java 11 which resulted in the Java 8 runs needing to compile twice as the compiler version changed. Unfortunately, I'm not explicitly certain on why this was done.

**Pull request merged**